### PR TITLE
Add replica set signup 

### PR DESCRIPTION
**Pull request by vicky-g**
### Description
This PR is necessary for assigning a replica set to users on signup. 

Changes include:
- updating the if check to see if a user has a `creator_node_endpoint` field instead of the `is_creator` flag (as part of the move to eventually deprecate relying on the `is_creator` flag)
- only using the dapp selection service if the current user does not have a `creator_node_endpoint` 

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

Manually tested on dapp to make sure dapp still works as expected

### CODE REVIEW

File: src/containers/upload-page/store/sagas.js [hunk 1]
```diff
@@ -1001,7 +1001,9 @@ function* uploadTracksAsync(action) {
     )
   )
 
-  if (!user.is_creator || !user.creator_node_endpoint) {
+  // If user already has creator_node_endpoint, do not reselct replica set
+  let newEndpoint = ''
+  if (!user.creator_node_endpoint) {
```
Comment (SidSethi) [inline src/containers/upload-page/store/sagas.js:1006]: tracks can only be uploaded if user is not a creator right? why remove that check

**Review by SidSethi — COMMENTED** [review]
can you summarize the changes inline or in PR description?

File: src/containers/upload-page/store/sagas.js (hunk 1 above)
Comment (vicky-g) [reply src/containers/upload-page/store/sagas.js:1006]: didn't we decide that we were going to keep the `is_creator` flag, but not actually use it in the code base? that was the motive behind this code change

File: src/containers/upload-page/store/sagas.js [hunk 2]
```diff
@@ -1001,7 +1001,9 @@ function* uploadTracksAsync(action) {
     )
   )
 
-  if (!user.is_creator || !user.creator_node_endpoint) {
+  // If user already has creator_node_endpoint, do not reselct replica set
```
Comment (piazzatron) [inline src/containers/upload-page/store/sagas.js:1004]: type `reselct`

File: src/containers/upload-page/store/sagas.js [hunk 3]
```diff
@@ -1031,32 +1033,41 @@ function* uploadTracksAsync(action) {
...
-    try {
-      console.debug(`Attempting to upgrade user ${user.user_id} to creator`)
+  // Try to upgrade to creator, early return if failure
+  let metadata = { is_creator: true }
+  try {
+    console.error(`Attempting to upgrade user ${user.user_id} to creator`)
+    if (newEndpoint && newEndpoint.length > 0) {
+      // If new endpoint was selected via the selection logic above, update
+      // the user's creator_node_endpoint and is_creator flag
+      console.error(`Upgrading user ${user.user_id} to creator...`)
```
Comment (piazzatron) [inline src/containers/upload-page/store/sagas.js:1046]: console.error?

File: src/containers/upload-page/store/sagas.js [hunk 4]
```diff
@@ -1031,32 +1033,41 @@ function* uploadTracksAsync(action) {
...
       yield call(AudiusBackend.upgradeToCreator, newEndpoint)
-    } catch (err) {
-      console.error(`Upgrade to creator failed with error: ${err}`)
-      yield put(uploadActions.uploadTrackFailed())
-      yield put(uploadActions.upgradeToCreatorError(err))
-      return
+      metadata['creator_node_endpoint'] = newEndpoint
```
Comment (raymondjacobson) [inline src/containers/upload-page/store/sagas.js:1048]: nit, metadata.creator_node_endpoint is usually the style we do

File: src/services/AudiusBackend.js [hunk 5]
```diff
@@ -817,7 +817,18 @@ class AudiusBackend {
     }
   }
 
-  static async upgradeToCreator(newCreatorNodeEndpoint) {
+  /**
+   * Updates a user's is_creator metadata field to true
+   */
+  static async updateIsCreatorFlagToTrue () {
+    audiusLibs.User.updateIsCreatorFlagToTrue()
```
Comment (raymondjacobson) [inline src/services/AudiusBackend.js:824]: what's this? i dont think we should include this -- we should just stop relying on is_creator or maybe set it implicitly in libs. introducing the client dependency on this function seems debty

File: src/containers/upload-page/store/sagas.js (hunk 3 above)
Comment (vicky-g) [reply src/containers/upload-page/store/sagas.js:1046]: ah, I wanted the console logs to stick out when I was viewing them in the browser. will update. is console logging ok or should we not log anything?

File: src/services/AudiusBackend.js [hunk 6]
```diff
@@ -1425,7 +1436,7 @@ class AudiusBackend {
       email,
       password,
       metadata,
-      metadata.is_creator /* false */,
+      audiusLibs.ServiceProvider,
```
Comment (vicky-g) [inline src/services/AudiusBackend.js:1439]: remove this

File: src/containers/upload-page/store/sagas.js (hunk 1 above)
Comment (SidSethi) [reply src/containers/upload-page/store/sagas.js:1006]: nvm you right

File: src/containers/upload-page/store/sagas.js [hunk 7]
```diff
@@ -1031,32 +1033,33 @@ function* uploadTracksAsync(action) {
...
+    yield put(uploadActions.uploadTrackFailed())
+    yield put(uploadActions.upgradeToCreatorError(err))
+    return
   }
 
+  yield put(
+    cacheActions.update(Kind.USERS, [
+      {
+        id: user.user_id,
+        metadata
```
Comment (raymondjacobson) [inline src/containers/upload-page/store/sagas.js:1058]: you could revert the change here and just have this set both

**Review by raymondjacobson — APPROVED** [review]

**Comment by raymondjacobson** [comment]
I think you have some lint errors

File: src/containers/upload-page/store/sagas.js [hunk 8]
```diff
@@ -1031,32 +1033,33 @@ function* uploadTracksAsync(action) {
...
-        }
-      ])
-    )
+  // Try to upgrade to creator, early return if failure
+  let metadata = { is_creator: true }
+  try {
+    // If new endpoint was selected via the selection logic above, update
+    // the user's creator_node_endpoint and is_creator flag
+    console.debug(`Attempting to upgrade user ${user.user_id} to creator`)
+    yield call(AudiusBackend.upgradeToCreator, newEndpoint)
```
Comment (SidSethi) [inline src/containers/upload-page/store/sagas.js:1045]: is it worth adding a check to error if `newEndpoint` is not set at this point?

Comment (vicky-g) [reply src/containers/upload-page/store/sagas.js:1045]: newEndpoint should always be set to the actual replica set assigned to the user. updated so that newEndpoint always has a non-empty value

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/vg-add-replica-set-signup-audbackendchanges

**Review by raymondjacobson — APPROVED** [review]

File: package.json [hunk 9]
```diff
@@ -6,7 +6,7 @@
   "version": "0.24.0",
   "private": true,
   "dependencies": {
-    "@audius/libs": "1.1.10",
+    "@audius/libs": "^1.1.11",
```
Comment (raymondjacobson) [inline package.json:9]: can you make this not ^

**Review by raymondjacobson — CHANGES_REQUESTED** [review]

File: package.json (hunk 9 above)
Comment (SidSethi) [reply package.json:9]: good catch

**Review by raymondjacobson — APPROVED** [review]
